Source/Core/ShaderConductor.cpp

```diff
@@ -305,6 +320,129 @@ namespace
         result.hasError = true;
     }
 
+#ifdef LLVM_ON_WIN32
+    template <typename T>
+    static HRESULT CreateDxcReflectionFromBlob(IDxcBlob* dxilBlob, CComPtr<T>& outReflection)
+    {
+        IDxcContainerReflection* containReflection = Dxcompiler::Instance().ContainerReflection();
+        IFT(containReflection->Load(dxilBlob));
+
+        uint32_t dxilPartIndex = ~0u;
+        IFT(containReflection->FindFirstPartKind(hlsl::DFCC_DXIL, &dxilPartIndex));
+        HRESULT result = containReflection->GetPartReflection(dxilPartIndex, __uuidof(*outReflection), (void**)(&outReflection));
+
+        return result;
+    }
+
+    void ShaderReflection(IDxcBlob* dxilBlob, Compiler::ResultDesc* results)
+    {
+        CComPtr<ID3D12ShaderReflection> shaderReflection;
+        IFT(CreateDxcReflectionFromBlob(dxilBlob, shaderReflection));
+
+        D3D12_SHADER_DESC shaderDesc;
+        shaderReflection->GetDesc(&shaderDesc);
+
+        D3D12_SHADER_INPUT_BIND_DESC bindDesc;
+
+        static std::vector<Compiler::ReflectionDesc> vecReflectionDescs;
```

Comment:
A better way is make reflectionDescs a blob, allocate memory for it, and deallocate it when destroy the result. Static can create a lot of troubles.